pkg/chain/ethlike/config.go

```diff
@@ -0,0 +1,74 @@
+package ethlike
+
+import "math/big"
+
+// Account is a struct that contains the configuration for accessing an
+// ETH-like network and a contract on the network.
+type Account struct {
+	// Keyfile is a full path to a key file.  Normally this file is one of the
+	// imported keys in your local chain node.  It can normally be found in
+	// a directory <some-path>/data/keystore/ and starts with its creation date
+	// "UTC--.*".
+	KeyFile string
+
+	// KeyFilePassword is the password used to unlock the account specified in
+	// KeyFile.
+	KeyFilePassword string
+}
+
+// CommonConfig is a struct that contains the configuration needed to connect
+// to an ETH-like node. This information will give access to an ETH-like network.
+type CommonConfig struct {
+	// Example: "ws://192.168.0.157:8546".
+	URL string
+
+	// Example: "http://192.168.0.157:8545".
+	URLRPC string
+
+	// A  map from contract names to contract addresses.
+	ContractAddresses map[string]string
+
+	Account Account
+
+	// MiningCheckInterval is the interval in which transaction
+	// mining status is checked. If the transaction is not mined within this
+	// time, the gas price is increased and transaction is resubmitted.
+	MiningCheckInterval int
+
+	// MaxGasPrice specifies the maximum gas price the client is
+	// willing to pay for the transaction to be mined. The offered transaction
+	// gas price can not be higher than the max gas price value. If the maximum
+	// allowed gas price is reached, no further resubmission attempts are
+	// performed.
+	MaxGasPrice *Value
+
+	// RequestsPerSecondLimit sets the maximum average number of requests
+	// per second which can be executed against the ETH-like node.
+	// All types of chain requests are rate-limited,
+	// including view function calls.
+	RequestsPerSecondLimit int
+
+	// ConcurrencyLimit sets the maximum number of concurrent requests which
+	// can be executed against the ETH-like node at the same time.
+	// This limit affects all types of chain requests,
+	// including view function calls.
+	ConcurrencyLimit int
+
+	// BalanceAlertThreshold defines a minimum value of the operator's
+	// account balance below which an alert will be triggered.
+	BalanceAlertThreshold *Value
+}
+
+// BalanceAlertThresholdValue returns the `BalanceAlertThreshold` integer value.
+func (cg *CommonConfig) BalanceAlertThresholdValue() *big.Int {
+	if cg.BalanceAlertThreshold != nil {
+		return cg.BalanceAlertThreshold.Int
+	}
+
+	return nil
+}
+
+// GetAccount returns the account configuration.
+func (cg *CommonConfig) GetAccount() Account {
+	return cg.Account
+}
```

Comment:
There is an interface introduced by https://github.com/keep-network/keep-ecdsa/pull/686 which makes use of that method.

Reply:
Do we really need `chainConfig` interface? We have two separate connectors in keep-network/keep-ecdsa#686 and each one of them can read from a specified config type. Actually, they already do it, for example: `config.Celo.Account.KeyFile` in `connector_celo.go`.

Reply:
It's not for connectors. It is needed by the `start.go` command which can treat the config in a chain-agnostic way. For example, it can run the balance monitoring.

Reply:
Depending on the result of the discussion about [how the price is represented for balance monitoring discussion](https://github.com/keep-network/keep-common/pull/66#discussion_r580153402) we may be able to get rid of it altogether.

Reply:
See https://github.com/keep-network/keep-common/commit/e715d7f415d40236121dd71ac20d8e4e33cf3c0b